### ISSUE TITLE
Errors after upgrade to v1.4.2

### ISSUE THREAD

**Issue by lordscarlet** [nathanwoulfe/Plumber-2#56]
**Describe the bug**
Several different errors after upgrade from `1.3.9-beta-000415` to `1.4.2`. Included are:

Navigating to `Approval Groups` gets the following error when calling `getAll`:

```javascript
{
  "Message": "No HTTP resource was found that matches the request URI 'https://localhost/umbraco/backoffice/Plumber/Groups/getAll'.",
  "MessageDetail": "No action was found on the controller 'Groups' that matches the name 'getAll'."
}
```

Cancelling an active workflow, on the `cancel` async call gets:

```javascript
{
  "Message": "An error has occurred.",
  "ExceptionMessage": "Sequence contains no elements",
  "ExceptionType": "System.InvalidOperationException",
  "StackTrace": "   at System.Linq.Enumerable.First[TSource](IEnumerable`1 source)\r\n   at Plumber.Core.Repositories.Implement.BaseWorkflowRepository`1.Single(String query, Expression`1 predicate, Object[] args) in D:\\a\\1\\s\\Plumber.Core\\Repositories\\Implement\\WorkflowRepository.cs:line 61\r\n   at Plumber.Core.Services.Implement.InstancesService.GetByGuid(Guid guid) in D:\\a\\1\\s\\Plumber.Core\\Services\\Implement\\InstancesService.cs:line 287\r\n   at Plumber.Web.Api.ActionsController.CancelWorkflowTask(TaskData model) in D:\\a\\1\\s\\Plumber.Web\\Api\\ActionsController.cs:line 234\r\n   at lambda_method(Closure , Object , Object[] )\r\n   at System.Web.Http.Controllers.ReflectedHttpActionDescriptor.ActionExecutor.<>c__DisplayClass6_2.<GetExecutor>b__2(Object instance, Object[] methodParameters)\r\n   at System.Web.Http.Controllers.ReflectedHttpActionDescriptor.ExecuteAsync(HttpControllerContext controllerContext, IDictionary`2 arguments, CancellationToken cancellationToken)\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   at System.Web.Http.Controllers.ApiControllerActionInvoker.<InvokeActionAsyncCore>d__1.MoveNext()\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   at System.Web.Http.Filters.ActionFilterAttribute.<CallOnActionExecutedAsync>d__6.MoveNext()\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n   at System.Web.Http.Filters.ActionFilterAttribute.<CallOnActionExecutedAsync>d__6.MoveNext()\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   at System.Web.Http.Filters.ActionFilterAttribute.<ExecuteActionFilterAsyncCore>d__5.MoveNext()\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   at System.Web.Http.Filters.ActionFilterAttribute.<CallOnActionExecutedAsync>d__6.MoveNext()\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n   at System.Web.Http.Filters.ActionFilterAttribute.<CallOnActionExecutedAsync>d__6.MoveNext()\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   at System.Web.Http.Filters.ActionFilterAttribute.<ExecuteActionFilterAsyncCore>d__5.MoveNext()\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   at System.Web.Http.Filters.ActionFilterAttribute.<CallOnActionExecutedAsync>d__6.MoveNext()\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n   at System.Web.Http.Filters.ActionFilterAttribute.<CallOnActionExecutedAsync>d__6.MoveNext()\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   at System.Web.Http.Filters.ActionFilterAttribute.<ExecuteActionFilterAsyncCore>d__5.MoveNext()\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   at System.Web.Http.Filters.ActionFilterAttribute.<CallOnActionExecutedAsync>d__6.MoveNext()\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n   at System.Web.Http.Filters.ActionFilterAttribute.<CallOnActionExecutedAsync>d__6.MoveNext()\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   at System.Web.Http.Filters.ActionFilterAttribute.<ExecuteActionFilterAsyncCore>d__5.MoveNext()\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   at System.Web.Http.Controllers.ActionFilterResult.<ExecuteAsync>d__5.MoveNext()\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   at System.Web.Http.Filters.AuthorizationFilterAttribute.<ExecuteAuthorizationFilterAsyncCore>d__3.MoveNext()\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   at System.Web.Http.Filters.AuthorizationFilterAttribute.<ExecuteAuthorizationFilterAsyncCore>d__3.MoveNext()\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   at System.Web.Http.Filters.AuthorizationFilterAttribute.<ExecuteAuthorizationFilterAsyncCore>d__3.MoveNext()\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   at System.Web.Http.Controllers.ExceptionFilterResult.<ExecuteAsync>d__6.MoveNext()\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n   at System.Web.Http.Controllers.ExceptionFilterResult.<ExecuteAsync>d__6.MoveNext()\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   at System.Web.Http.Dispatcher.HttpControllerDispatcher.<SendAsync>d__15.MoveNext()"
}
```

which then tries to call `/umbraco/backoffice/Plumber/Tasks/tasksbyinstanceguid/undefined`

**To Reproduce**

We have an unusual setup that does not pull Umbraco from nuget. We have a copy of the `v8.8` source that we compile and place some dependencies on top of. We essentially have 8.8, but I can't say for sure that our other dependencies don't somehow cause problems.

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Browser Edge (Chromium)
 - Version 86.0.622.63

**Plumber version**
1.4.2

**Additional context**
As mentioned, we don't have a 100% vanilla configuration, but it should be pretty close.

**Comment by nathanwoulfe** [comment]
Hey @lordscarlet, this is all a bit odd - the GetAll method definitely exists, works fine in my local (umb 8.8, latest plumber codebase). Can you try requesting `/umbraco/backoffice/Plumber/Groups/get/2` where 2 is the group id - if there's no group, you'll get a SQL error, but we really just need to verify you can hit the endpoint.

The undefined guid is strange too, since it must exist if workflow is visible in the backoffice - without it, there's no way to have fetched the task info to then render the cancel button. Is the workflow valid ie has tasks, isn't completed, hasn't got stuck in a different state?

I'd try deleting the app_plugins/plumber folder, and the plumber.core and plumber.web dlls, then reinstalling. It could be a bit of strange cache action, since there was a fair bit of change in 1.4.0. There's a current 1.4.3 nightly available too, which will be a proper release later this week

**Comment by lordscarlet** [comment]
That returned what I assume is expected: 

```javascript
{
   "description":null,
   "alias":"testGroup",
   "groupEmail":null,
   "offlineApproval":false,
   "deleted":false,
   "permissions":[
      
   ],
   "users":[
      
   ],
   "usersSummary":"||",
   "inheritMembers":null,
   "groupId":1,
   "name":"TestGroup"
}
```

And another with more data:

```javascript
{
   "description":null,
   "alias":"submitBug",
   "groupEmail":null,
   "offlineApproval":false,
   "deleted":false,
   "permissions":[
      {
         "id":0,
         "nodeId":43652,
         "groupId":2,
         "groupName":"Submit Bug",
         "contentTypeId":0,
         "permission":0,
         "condition":null,
         "variant":"en-US",
         "type":0,
         "nodeName":"Sandbox",
         "contentTypeName":null,
         "contentTypeAlias":null,
         "userGroup":null
      }
   ],
   "users":[
      {
         "userId":3,
         "groupId":2,
         "username":"xxx",
         "name":"Douglas Moore",
         "email":"xxx@xxx.xxx",
         "inherited":false,
         "isActive":true
      }
   ],
   "usersSummary":"|3|",
   "inheritMembers":null,
   "groupId":2,
   "name":"Submit Bug"
}
```

FWIW, one of our other devs was having the issue, and that's what pulled me into it. I then separately installed it on my local, and I saw the same issue as them.

**Comment by lordscarlet** [comment]
@nathanwoulfe I am definitely not discounting the possibility of something odd/conflicting in our configuration

**Comment by nathanwoulfe** [comment]
@lordscarlet I think you may be right - I've installed v1.4.3 onto a clean v8.8 site and it works fine... I'd try upgrading to the latest nightly, which has improved logging which may (or may not) give some better insights...

**Comment by lordscarlet** [comment]
@nathanwoulfe I loaded `1.4.3-beta-000479` and... no errors!

**Comment by lordscarlet** [comment]
..the other member of my team, however, still getting the error(s). Will dig some more.

**Comment by lordscarlet** [comment]
@nathanwoulfe I noticed something different between myself and the non-working developer setup. Base don DevTools it looks like my instance is not calling `/umbraco/backoffice/Plumber/Groups/getAll` at all. But, if I dig deeper, it is, but it is calling `/umbraco/backoffice/Plumber/Groups/getAll/false`. If the non-working installation calls it with `/false`, they get a proper response.

**Comment by lordscarlet** [comment]
@nathanwoulfe [insert smashing things GIF]

The solution was to increment the version in `ClientDependency.config`

**Comment by nathanwoulfe** [comment]
Good old client dependency... Backoffice caching can be a pain in the ass. 

Glad to hear it's resolved though, I'll get 1.4.3 out today.
